### PR TITLE
Find common root directory

### PR DESCRIPTION
**Pull request by WearyMonkey**

### CODE REVIEW

File: internal/ibazel/ibazel.go [hunk 1]
```diff
@@ -532,21 +532,29 @@ func (i *IBazel) watchFiles(query string, watcher common.Watcher) {
 		return
 	}
 
+	rootParent, _ := filepath.Split(toWatch[0])
+	rootSplit := strings.Split(strings.Trim(rootParent, "/"), "/")
```
Comment (uri-canva) [inline internal/ibazel/ibazel.go]: https://pkg.go.dev/path/filepath#SplitList
```suggestion
	rootSplit := filepath.SplitList(rootParent)
```

File: internal/ibazel/ibazel.go [hunk 2]
```diff
@@ -532,21 +532,29 @@ func (i *IBazel) watchFiles(query string, watcher common.Watcher) {
...
 
 	for _, file := range toWatch {
 		if _, err := os.Stat(file); !os.IsNotExist(err) {
 			filesWatched[file] = struct{}{}
 		}
-
-		parentDirectory, _ := filepath.Split(file)
-		// Add a watch to the file's parent directory, we might already have this dir in our set but thats OK
-		uniqueDirectories[parentDirectory] = struct{}{}
+		split := strings.Split(strings.Trim(file, "/"), "/")
```
Comment (uri-canva) [inline internal/ibazel/ibazel.go]:
```suggestion
		split := filepath.SplitList(file)
```

File: internal/ibazel/ibazel.go [hunk 3]
```diff
@@ -532,21 +532,29 @@ func (i *IBazel) watchFiles(query string, watcher common.Watcher) {
 		return
 	}
 
+	rootParent, _ := filepath.Split(toWatch[0])
+	rootSplit := strings.Split(strings.Trim(rootParent, "/"), "/")
+	rootLength := len(rootSplit)
 	filesWatched := map[string]struct{}{}
-	uniqueDirectories := map[string]struct{}{}
```
Comment (uri-canva) [inline internal/ibazel/ibazel.go]: You probably want to keep this so you don't do the common prefix extraction for every single file path, but instead only do it once for every unique directory path.

**Review by uri-canva — APPROVED** [review]

File: internal/ibazel/ibazel.go (hunk 1 above)
Comment (WearyMonkey) [reply internal/ibazel/ibazel.go]: I started with split list as well, but it actually does something different. From the example:

```
fmt.Println("On Unix:", filepath.SplitList("/a/b/c:/usr/bin"))
```

```
On Unix: [/a/b/c /usr/bin]
```

Comment (uri-canva) [reply internal/ibazel/ibazel.go]: Oh yeah you're right, the docs do say that. I guess `SplitList` means `Split applied to path lists`, not `Split returning a list`.

**Comment by WearyMonkey** [comment]
closed in favour of: https://github.com/bazelbuild/bazel-watcher/pull/559

**Pull request closed without merging**